Fb 4713 assay refactor

labkey-adam opened the pull request.

labkey-jeckels commented on assay/api-src/org/labkey/api/exp/property/AssayDomainKind.java:
```diff
@@ -1,147 +1,147 @@
-/*
```
Line ending change for entire file, would be nice to avoid.

labkey-klum replied:
One thing you could try for these renamed files, is to change the path to match the entry in the .gitattributes file so that we can continue to allow the CR/LF endings that were committed when we did the initial migration. The entry in the .gitattributes file allows an exception on a per file basis for the global line ending rule (which is LF).

labkey-adam replied:
I updated the entries for NabUrls and AssayDomainKind, and it didn't seem to make a difference in the GitHub diffs. Probably a good idea to revise the hundreds of files that got moved (ugh) for future revision purposes, though.

labkey-jeckels commented on assay/api-src/org/labkey/api/nab/NabUrls.java:
```diff
@@ -1,33 +1,33 @@
-/*
```
Line ending change for entire file, would be nice to avoid.

labkey-adam commented:
I don't know much... but from reading various posts (e.g., https://stackoverflow.com/questions/2314652/is-it-possible-to-move-rename-files-in-git-and-maintain-their-history), I'm not sure line endings are at fault here. It appears that (unlike SVN) Git doesn't track renames or moves at all, rather, various log/merge tools detect renames & moves after the fact (or they don't). I don't know why the GitHub viewer recognizes some of the renames but not others. However, when I view NabUrls and AssayDomainKind (post move, commit, push to FB) in IntelliJ, I see the full history and line-by-line annotations. Not sure there's anything we can do to make GitHub behave better with file moves.

labkey-adam commented:
Seems like CODEOWNERS should have been deleted by my recent merge from develop, but I deleted it here.

Also updated .gitattributes with hundreds of changes. #tedious
